### PR TITLE
feature: add login component

### PR DESCRIPTION
**Pull request by anwyss**

### CODE REVIEW

File: src/redux/action-creators/authentication.ts [hunk 1]
```diff
@@ -7,24 +7,19 @@ export interface SignUpCredentials {
...
-// interface LogInAction {
-//   type: "LOG_IN";
-//   payload: LogInCredentials;
-// }
-
-// interface LogOutAction {
-//   type: "LOG_OUT";
-// }
+export interface LogInAction {
+  type: "LOG_IN";
```
Comment (guillem-gelabert) [inline src/redux/action-creators/authentication.ts]: we already have this string as constant, we should avoid "magic strings"

File: src/redux/constants.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ export const SIGN_UP = "SIGN_UP";
 export const SIGN_UP_SUCCESS = "SIGN_UP_SUCCESS";
```
Comment (guillem-gelabert) [inline src/redux/constants.ts:2]: add SIGN_UP constant

File: src/redux/epics/login.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import { LOG_IN, LOG_IN_SUCCESS, LOG_IN_FAILURE } from "../constants";
+import client from "../../graphql/client";
+import { logIn } from "../../graphql/authentication";
+
+export const loginEpic: Epic = (action: Observable<LogInAction>) => {
+  return action.pipe(
+    ofType(LOG_IN),
+    mergeMap((currentAction: LogInAction) => {
+      return from(
+        client.query<any>({
```
Comment (guillem-gelabert) [inline src/redux/epics/login.ts]: I think this should be the return value of the query

**Review by guillem-gelabert — CHANGES_REQUESTED** [review]

File: src/redux/constants.ts (hunk 2 above)
Comment (anwyss) [reply src/redux/constants.ts:2]: it is there

Comment (guillem-gelabert) [reply src/redux/constants.ts:2]: my bad

**Review by guillem-gelabert — APPROVED** [review]

**Pull request merged**